plugins/plugin-pullrequest/codenvy-plugin-pullrequest-ide/src/main/java/com/codenvy/plugin/pullrequest/client/ContributionExtension.java

```diff
@@ -60,26 +51,19 @@
 @Extension(title = "Contributor", version = "1.0.0")
 public class ContributionExtension {
 
-    private final ContributePartPresenter   contributionPartPresenter;
     private final WorkflowExecutor          workflowExecutor;
-    private final VcsHostingServiceProvider hostingServiceProvider;
-    private final VcsServiceProvider        vcsServiceProvider;
 
-    private String  lastSelectedProjectName;
-    private boolean partWasOpened;
+    private Project lastSelectedProject;
 
     @Inject
     public ContributionExtension(final EventBus eventBus,
                                  final ContributeResources resources,
                                  final AppContext appContext,
-                                 final ContributePartPresenter contributionPartPresenter,
                                  final WorkflowExecutor workflow,
                                  final VcsHostingServiceProvider vcsHostingServiceProvider,
-                                 final VcsServiceProvider vcsServiceProvider) {
+                                 final ContributePartPresenter contributePart,
+                                 WorkspaceAgent workspaceAgent) {
```

Comment:
For what? This variable is not using in anonymous classes. So we don't need to mark it as final.